test/sql/include/deparse_create.sql

```diff
@@ -0,0 +1,45 @@
+-- This file and its contents are licensed under the Apache License 2.0.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-APACHE for a copy of the license.
+
+-- Lets create some tabels that we will try to deparse and recreate
+
+CREATE TABLE table1(time TIMESTAMP, v FLOAT8, c CHAR(10), x NUMERIC(10,4), i interval hour to minute);
+
+CREATE TABLE table2(time TIMESTAMP NOT NULL, v FLOAT8[], d TEXT COLLATE "fr_FR", num INT DEFAULT 100);
+
+CREATE TABLE table3(time TIMESTAMP PRIMARY KEY, v FLOAT8[][], num INT CHECK (num > 0), d INT UNIQUE, CONSTRAINT validate_num_and_d CHECK ( num > d));
+
+CREATE TABLE table4(t TIMESTAMP , d INT, PRIMARY KEY (t, d)); 
+
+CREATE TABLE ref_table(id INT PRIMARY KEY, d TEXT); 
+
+CREATE TABLE table5(t TIMESTAMP PRIMARY KEY, v FLOAT8, d INT REFERENCES ref_table ON DELETE CASCADE);
+
+CREATE SEQUENCE my_seq;
+
+CREATE UNLOGGED TABLE table6(id INT NOT NULL DEFAULT nextval('my_seq'), t TEXT);
+
+CREATE INDEX ON table6 USING BTREE (t);
+
+\c :TEST_DBNAME :ROLE_SUPERUSER
```

Comment:
Do this at the top of the test.
Then use `SET ROLE :ROLE_DEFAULT_PERM_USER` and `RESET ROLE` to switch users. This avoids creating new connections.